src/controllers/image.controller.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { Request, response, Response } from "express";
+import imageService, { ImageService } from "../services/image.service";
+import sharp from "sharp";
+import { matchedData } from "express-validator";
+
+export class ImageController {
+  constructor(private imageService: ImageService) {}
+
+  public async addImage(req: Request, res: Response): Promise<Response> {
+    try {
+      const { _id, isAdmin } = req.user!;
+      const recipeId: string = matchedData(req).id;
+      const image = await sharp(req.file?.buffer)
+        .resize({ width: 250, height: 250 })
+        .png()
+        .toBuffer();
+      const savedImage = await imageService.addRecipeImage(
+        recipeId,
+        image,
+        _id,
+        isAdmin
+      );
+      if (savedImage) {
+        return res.status(201).send();
```

Comment:
It's ok to do it this way, but since we're agreed on using `http-status-codes` library, please use it everywhere you type a HTTP status code